types/index.d.ts

```diff
@@ -172,22 +271,81 @@ export namespace Service {
 }
 
 export namespace streams {
-  class BlockDecoder {
-    constructor(opts?: any);
-    static defaultCodecs(): any;
+
+  class BlockDecoder extends stream.Duplex {
+    constructor(opts?: Partial<DecoderOptions>);
+    static defaultCodecs(): CodecOptions;
+
+    //adding all listeners, etc. Can't add a single method override for metadata, all base methods need to be repeated :-(
```

Comment:
Are these really required? How come they aren't provided by `Duplex`?

Reply:
The implementation of most of these is provided by Duplex. However, in the definition files, Typescipt requires you to specify the overload function (the one with string, (...args:any[]) ). Once you do that, all the events from the base-class (Duplex) are hidden from code completion (at least in vscode). To make them visible again, you have to add them all :-(

I see this as a Typescript or otherwise vscode bug...

Reply:
That's a bummer. It might be surprising to add them here but not to the other duplex streams below; I'd prefer staying consistent and not adding them anywhere, at the cost of losing auto-completion (until the issue you mention is fixed) - do you mind you take them out? (Adding them everywhere would be an unreasonable amount of clutter for the benefits.)